lib/av_client/crypto/pedersen_commitment.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  addPoints,
+  Bignum,
+  bignumFromHex,
+  bignumToHex,
+  Curve,
+  generateRandomBignum,
+  hashToBignum,
+  Point,
+  pointFromHex,
+  pointFromX,
+  pointToHex,
+} from "./util";
+
+class PedersenCommitment {
+  static computeGenerator(index) {
+    const baseGeneratorPrefix = () => pointToHex(new Point(Curve.G));
+    const secp256k1_curve_prime = new Bignum(Curve.field.modulus);
+
+    const target = [baseGeneratorPrefix(), index].join(',');
+    let x = hashToBignum(target).mod(secp256k1_curve_prime);
+
+    /*eslint no-constant-condition: ["error", { "checkLoops": false }]*/
+    while(true) {
+      try {
+        return pointFromX(x);
+      }
+      catch(err) {
+        // Skip to next iteration
```

Comment:
Please explain why this is ok.

Reply:
Same as above. My understanding is that there's a 50% chance that a point will be found in a given iteration. SJCL will throw an error if an x-value does not correspond to a valid point on the curve.

Looking at it again, I see that there is of course the possibility that "some other" error occurs, which cannot be alleviated by incrementing the x-value -> Possibly leading to an infinite loop.

I will allow it to only continue for that specific error - and rethrow any other error.

Reply:
Yes, but even if we use a catch all, please explain why the `catch` is a no-op.

Reply:
Got it. See this commit 4c27e4a.